staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -342,7 +370,11 @@ func (t *tracker) Delete(gvr schema.GroupVersionResource, ns, name string) error
 			return err
 		}
 		if objMeta.GetNamespace() == ns && objMeta.GetName() == name {
```

Comment:
This doesn't seem to be handling the "empty namespace is all namespaces" case.

I think you may need a test for your test harness.